packages/mui-base/src/Modal/Modal.types.ts

```diff
@@ -20,6 +20,9 @@ export interface ModalOwnProps {
    * An HTML element or function that returns one.
    * The `container` will have the portal children appended to it.
    *
+   * You can also provide a callback, which is called in a React layout effect.
+   * It allows to set the container from a ref or to not break server-side rendering.
```

Comment:
This sentence was tough to understand at first—here's how I'd phrase it (if I'm understanding correctly):

```suggestion
   * This lets you set the container from a ref, and also makes server-side rendering possible.
```